exporters/otlp/all/src/main/java/io/opentelemetry/exporter/otlp/internal/OtlpGrpcSpanExporterComponentProvider.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.otlp.internal;
+
+import static io.opentelemetry.exporter.otlp.internal.OtlpConfigUtil.DATA_TYPE_TRACES;
+
+import io.opentelemetry.api.incubator.config.DeclarativeConfigProperties;
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter;
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporterBuilder;
+import io.opentelemetry.sdk.autoconfigure.spi.internal.ComponentProvider;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+
+/**
+ * Declarative configuration SPI implementation for {@link OtlpGrpcSpanExporter}.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ */
+public class OtlpGrpcSpanExporterComponentProvider implements ComponentProvider<SpanExporter> {
+
+  @Override
+  public Class<SpanExporter> getType() {
+    return SpanExporter.class;
+  }
+
+  @Override
+  public String getName() {
+    return "otlp_grpc";
+  }
+
+  @Override
+  public SpanExporter create(DeclarativeConfigProperties config) {
+    OtlpGrpcSpanExporterBuilder builder = grpcBuilder();
+
+    OtlpDeclarativeConfigUtil.configureOtlpExporterBuilder(
+        DATA_TYPE_TRACES,
+        config,
+        builder::setEndpoint,
+        builder::addHeader,
+        builder::setCompression,
+        builder::setTimeout,
+        builder::setTrustedCertificates,
+        builder::setClientTls,
+        builder::setRetryPolicy,
+        builder::setMemoryMode,
+        /* isHttpProtobuf= */ false);
```

Comment:
Is this because we still default spans to gRPC in the sdk or something?

Reply:
`configureOtlpExporterBuidler` is a shared utility function which interprets standard OTLP exporter configuration options and calls these various consumers to set the correct options on the OTLP exporter builders.

This particular argument is used to validate the OTLP endpoint:
- If `isHttpProtobuf=false`, then the protocol is grpc, and we validate that the endpoint doesn't have a path (since path is not allowed in grpc) and log a warning if the port is not 4317
- If `isHttpProtobuf=true`, then we log a warning if the port is not 4318